src/template-tags/tickets.php

```diff
@@ -745,272 +767,486 @@ function tribe_tickets_has_meta_fields( $id, $context = null ) {
-		$html[] = ')';
+	/**
+	 * Builds and returns the correct ticket repository.
+	 *
+	 * @since 4.8
+	 *
+	 * @param string $repository The slug of the repository to build/return.
+	 *
+	 * @return Tribe__Repository__Interface
+	 */
+	function tribe_tickets( $repository = 'default' ) {
+		$map = [
+			'default'        => 'tickets.ticket-repository',
+			'rsvp'           => 'tickets.ticket-repository.rsvp',
+			'tribe-commerce' => 'tickets.ticket-repository.commerce',
+			'restv1'         => 'tickets.rest-v1.ticket-repository',
+		];
+
+		/**
+		 * Filters the map relating ticket repository slugs to service container bindings.
+		 *
+		 * @since 4.8
+		 *
+		 * @param array  $map        A map in the shape [ <repository_slug> => <service_name> ]
+		 * @param string $repository The currently requested implementation.
+		 */
+		$map = apply_filters( 'tribe_tickets_ticket_repository_map', $map, $repository );
+
+		return tribe( Tribe__Utils__Array::get( $map, $repository, $map['default'] ) );
 	}
+}
 
-	$html = implode( '', $html );
+if ( ! function_exists( 'tribe_attendees' ) ) {
 
-	if ( true === $display ) {
-		echo $html;
+	/**
+	 * Builds and returns the correct attendee repository.
+	 *
+	 * @since 4.8
+	 *
+	 * @param string $repository The slug of the repository to build/return.
+	 *
+	 * @return Tribe__Repository__Interface
+	 */
+	function tribe_attendees( $repository = 'default' ) {
+		$map = [
+			'default'        => 'tickets.attendee-repository',
+			'rsvp'           => 'tickets.attendee-repository.rsvp',
+			'tribe-commerce' => 'tickets.attendee-repository.commerce',
+			'restv1'         => 'tickets.rest-v1.attendee-repository',
+		];
+
+		/**
+		 * Filters the map relating attendee repository slugs to service container bindings.
+		 *
+		 * @since 4.8
+		 *
+		 * @param array  $map        A map in the shape [ <repository_slug> => <service_name> ]
+		 * @param string $repository The currently requested implementation.
+		 */
+		$map = apply_filters( 'tribe_tickets_attendee_repository_map', $map, $repository );
+
+		return tribe( Tribe__Utils__Array::get( $map, $repository, $map['default'] ) );
 	}
+}
+
+if ( ! function_exists( 'tribe_get_rsvp_label_singular' ) ) {
 
-	return $html;
+	/**
+	 * Get the singular version of the RSVP label. May also be used as a verb.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $context Allows passing additional context to this function's filter, e.g. 'verb' or 'template.php'.
+	 *
+	 * @return string
+	 */
+	function tribe_get_rsvp_label_singular( $context = '' ) {
+		/**
+		 * Allows customization of the singular version of the RSVP label.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $label   The singular version of the RSVP label. Defaults to "RSVP".
+		 * @param string $context The context in which this string is filtered, e.g. 'verb' or 'template.php'.
+		 */
+		return apply_filters( 'tribe_get_rsvp_label_singular', _x( 'RSVP', 'singular label for RSVP', 'event-tickets' ), $context );
```

Comment:
While it's very useful to use `__FUNCTION__` we should use the hardcoded string. Why? Because if the function name changes, if we move it into a class method, or if IDE kicks into usages of a filter, things start to get tricky.

I've already made the change for you but wanted to make sure you were aware going forward.